src/python/pants/core/goals/fmt.py

```diff
@@ -31,17 +31,21 @@ class FmtResult:
     digest: Digest
```

Comment:
I like that, I think. Do you have any concerns about the extra memory to store the `input: Digest` on every `FmtResult`, or likely premature optimization?

Reply:
`Digest`s are 40 bytes long, so you should never worry about holding them in memory.

A `Digest` is the in-memory representation of a snapshot stored in the LMDB database, so it does not force more than 40 bytes to be held in memory.

A `Snapshot` is slightly larger because it is a `Digest` + a list of filenames (ie, a bunch of strings), but still doesn't actually hold file content in memory.